src/minimax_server/router.clj

```diff
@@ -0,0 +1,18 @@
+(ns minimax-server.router
+  (:import [java.util.function Function]
+           [scarvill.httpserver.response ResponseBuilder])
+  (:require [clojure.string :as str]))
+
```

Comment:
When you're doing a lot of java calls in a namespace it's typically a good idea to add `(set! *warn-on-reflection* true)` just below the namespace declaration. Everything in Clojure is just of type `java.lang.Object` so it has to resort to the slower reflection API to resolve method calls. This will warn you when it's doing that so you can add the appropriate type hints.